### PR TITLE
initial commit, simple, separated models

### PR DESCRIPTION
**Pull request by snurkabill**
not sure, how this repository is planned to be. so this is just first attemt. feel free to criticise it

### CODE REVIEW

**Comment by martinwicke** [comment]
Sorry, my notification settings were messed up. I have to fix some CLAbot settings, and then I'll look at your PR in detail.

Thanks!

**Comment by martinwicke** [comment]
Can you say something? I'm checking that the CLA bot works.

**Comment by snurkabill** [comment]
something. :)

**Comment by snurkabill** [comment]
something. :)

**Comment by snurkabill** [comment]
Why do I need sign some CLA? I don't want to fill in my mobile number atc... #justWhyGoogle

**Comment by martinwicke** [comment]
The CLA is the only way for us to ensure that the code in this repository is actually licensed under Apache2. If we didn't have CLAs, it would be nominally Apache2, but if someone finds their code here, or thinks they do, our defense may be weak. Therefore we cannot accept contributions without a CLA -- it protects not only Google, but also everybody who uses this code.

Our CLA is very reasonable. I'd love to accept your contribution, but I cannot without the CLA.

**Comment by snurkabill** [comment]
I signed it!

**Comment by snurkabill** [comment]
here you go. but only because you are Google, guys :P

**Comment by snurkabill** [comment]
would you be interested in more sophisticated, objected oriented model? Or are these simple separated models great as PoC? 

I can add some more, parametrized models.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+autoencoder/MNIST_data/*
```
Comment (jendap) [inline .gitignore]: If you are adding .gitignore cosider adding *.pyc there and remove the pyc files from this PR (and squash it, so the never make it to the repo).

**Comment by martinwicke** [comment]
These models are great. Can you remove the pyc as Jan said, then we can merge.

@ilblackdragon, this is using sklearn "bare", maybe down the road we can merge with some of your examples?

**Comment by snurkabill** [comment]
sure, I will do it, sorry for late responses

**Comment by snurkabill** [comment]
What about now?

**Comment by martinwicke** [comment]
Looks good. Can you [rebase and squash](http://rebaseandsqua.sh)?

**Comment by snurkabill** [comment]
sure, done :+1:

**Comment by martinwicke** [comment]
Thank you!

**Pull request merged**